addons/website_form_project/static/src/js/website_form_project_editor.js

```diff
@@ -26,9 +26,37 @@ FormEditorRegistry.add('create_task', {
         name: 'project_id',
         type: 'many2one',
         relation: 'project.project',
-        required: true,
         string: _t('Project'),
     }],
 });
 
 });
+
+odoo.define('website_form_project.project_row', function (require) {
+    'use strict';
+    const _t = require('web.core')._t;
+
+    require('website_form_editor').options.registry.WebsiteFormEditor.include({
+        /**
+         * Wraps a project we-select in a we-row element, and adds a
+         * we-button linking to the Project module
+         *
+         * @param {Object} field
+         * @returns {HTMLElement}
+         */
+        _buildSelect: function (field) {
+            if (field.name === 'project_id') {
+                const icon = document.createElement('a');
+                icon.classList.add('fa', 'fa-fw', 'fa-plus')
+                icon.href = "/web#action=open_view_project_all";
+                const linkButton = document.createElement('we-button');
+                linkButton.title = _t("Create a new Project");
+                linkButton.append(icon);
+                const projectRow = document.createElement('we-row');
+                projectRow.append(this._super.apply(this, arguments));
+                projectRow.append(linkButton);
+                return projectRow;
+            }
+        },
+    });
```

Comment:
It could be done this way but obviously if you wrap the whole method in a IF and don't call super outside of that if... nothing will work anymore as @JKE-be explained :/

Making a generic feature as explained by @JKE-be seems to be a good idea. So something like doing this code which add a link button with a given URL directly in website_form and only passing the correct URL value from website_form_project, as a parameter for the option. That way if another module needs to do that, it won't have to recode this whole logic of button addition, it will simply need to define the act_window URL.

Also, as explained in Discord, it can't but just a link with an href. We talked about what is to be done (like SaveSnippet, request_save, etc etc).